decidim-api/lib/decidim/api/test/component_context.rb

```diff
@@ -42,3 +42,185 @@
     )
   end
 end
+
+shared_examples "with resource visibility" do
+  let(:process_space_factory) { :participatory_process }
+
+  context "when space is published" do
+    let!(:participatory_process) { create(process_space_factory, :published, :with_steps, organization: current_organization) }
+
+    context "when component is published" do
+      let!(:current_component) { create(component_factory, :published, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
+        let!(:current_user) { create(:user, :admin, :confirmed, organization: current_organization) }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+
+      context "when user is visitor" do
+        let!(:current_user) { nil }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+
+      context "when user is normal user" do
+        let!(:current_user) { create(:user, :confirmed, organization: current_organization) }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+    end
+
+    context "when component is not published" do
+      let!(:current_component) { create(component_factory, :unpublished, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
```

Comment:
We have more roles that we need to add for this and the other contexts: 

- space admin
- valuators

Reply:
done in bd1c077a94